lib/serialization/serialization.nit

```diff
@@ -134,6 +134,18 @@ interface Serializable
 	# The subclass change the default behavior, which will accept references,
 	# to force to always serialize copies of `self`.
 	private fun serialize_to_or_delay(v: Serializer) do v.serialize_reference(self)
+
+	# Create an instance of this class from the `deserializer`
+	#
+	# This constructor is refined by subclasses to correctly build their instances.
+	init from_deserializer(deserializer: Deserializer) do end
+end
+
+redef interface Object
+	# Is `self` the same as `other` in a serialization context?
+	#
+	# Used to determine if an object has already been serialized.
+	fun is_same_serialized(other: nullable Object): Bool do return is_same_instance(other)
```

Comment:
No... it's meant to be redefed in sub-classes in other modules and it is used by an outside class, so it can't be protected.